Unify page names, grid names, form names

Hlavtox opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Browse all backoffice and check page titles.
| Fixed ticket?     | 
| Related PRs       | 
| Sponsor company   | 

### How to test
- Enable all feature flags.
- Browse backoffice and see nothing is broken.

### Changes
- Fixed HTML syntax of some card headers.
- Unified grid names - they should be in plural.
- Unified editing forms card names - their names should be singular.
- Unified a place where `enableSidebar` is defined - in the controllers.
- Unified a place where `layoutTitle` is defined - in the controllers. Exception is the more complicated one on order page.
- Added and unified missing page names (`layoutTitle`) everywhere.
  - **Listings:** `Categories`, `Brands`, `Tax rules`
  - **Detail:** `Brand Studio design`, `Result of MySQLRequest`, `Customer Daniel Hlaváček`, `Cart #1548`
  - **New:** `New category`, `New brand`, `New tax rule`
  - **Editation:** `Editing category Women`, `Editing brand Studio design`, `Editing tax rule CZ 21 %`
  - _I followed an example from Customers section, where it's done somehow correctly. For titles with IDs I kept the original logic with a hashtag, used throughout the backoffice._
- Added specific layoutTitles to every controller so they don't use a fallback from left menu.
_____________________________

### Layout title changes

If the change is around `'layoutTitle' =>` and the filename ends with `...Controller.php`, it!s a page name. On almost all places except Customer and Category sections, they were not done or not unified at least.

![layoutTitle](https://user-images.githubusercontent.com/6097524/219432034-03bdb0c5-829d-41bd-8133-c68f59a7e1f3.jpg)

-------------------------

### Form title changes
If the file with the change is `form.html.twig`, it is a header of an editation form. There were some form titles that were in plural, although they are concerned with a single entity.

![form title](https://user-images.githubusercontent.com/6097524/219432181-5a27f600-c88e-42b9-a939-a0eb53bd046a.jpg)

-------------------------

### Grid title changes
Everything that is wrapped around `protected function getName()` and the filename contains `Factory` is a grid. There were only a few fixes, mostly singular/plural form.

![grid](https://user-images.githubusercontent.com/6097524/219432845-9db3cef4-ce31-4d4e-a475-b50aee984260.jpg)

Hlavtox committed:
Unify titles, card names, grid names

0x346e3730 commented:
Can't this PR be splitted to smaller ones ? Will be a pain to review

kpodemski commented on src/Core/Grid/Definition/Factory/MetaGridDefinitionFactory.php:
```diff
@@ -64,7 +64,7 @@ protected function getId()
     protected function getName()
     {
-        return $this->trans('SEO & URLs', [], 'Admin.Navigation.Menu');
+        return $this->trans('Page configuration', [], 'Admin.Navigation.Menu');
```
```suggestion
        return $this->trans('Pages configuration', [], 'Admin.Navigation.Menu');
```

🤔

Julievrz replied:
We don't think it's a good idea to replace the "SEO & URLs" wording with "Page configuration". A page with a similar name already exists in the Design section of the back office ("Pages").

Hlavtox replied:
TODO wording

Hlavtox replied:
TODO wording

kpodemski commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php:
```diff
@@ -153,6 +155,7 @@ function (array $language) {
                 'Admin.Notifications.Info'
             ),
             'multistoreIsUsed' => $this->get('prestashop.adapter.multistore_feature')->isUsed(),
+            'layoutTitle' => $this->trans('New status', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('New order state', 'Admin.Actions'),
```

Julievrz replied:
Let's use "status". It's more common. Here's a quick benchmark:
https://www.amazon.com/gp/help/customer/display.html?nodeId=201757550
https://help.shopify.com/en/manual/orders/order-status
https://faguo.zendesk.com/hc/en-150/articles/360008605820-What-does-the-status-of-my-order-mean-
https://help.etsy.com/hc/en-us/articles/115015521948-What-s-the-Status-of-My-Order-?segment=shopping

+ If it's a CTA, let's keep the "Add new X" structure for consistency.

Hlavtox replied:
New order status

kpodemski commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php:
```diff
@@ -186,18 +189,28 @@ public function editAction(int $orderStateId, Request $request)
                         'value' => sprintf('%s - %s', $language['iso_code'], $language['name']), ];
                 }, $this->get('prestashop.adapter.legacy.context')->getLanguages()),
+            'layoutTitle' => $this->trans(
+                'Editing status %name%',
```
```suggestion
                'Editing order state %name%',
```

I feel like we're missing something with status vs. state, it's almost the same but we should probably use one word?

Julievrz replied:
Daniel's suggestion is easier to scan. We're already on the order status page, so it's clear enough.

Hlavtox replied:
Status ✅

Hlavtox replied:
Editing order status %name%

kpodemski commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php:
```diff
@@ -358,6 +366,7 @@ public function createCmsCategoryAction(Request $request)
                 'cmsPageCategoryForm' => $cmsPageCategoryForm->createView(),
                 'enableSidebar' => true,
                 'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+                'layoutTitle' => $this->trans('New page category', 'Admin.Actions'),
```
```suggestion
                'layoutTitle' => $this->trans('New category for pages', 'Admin.Actions'),
```

Julievrz replied:
Daniel's suggestion looks good, but it needs a more specific translation domain.

kpodemski commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php:
```diff
@@ -423,6 +432,13 @@ public function editCmsCategoryAction($cmsCategoryId, Request $request)
                 'enableSidebar' => true,
                 'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+                'layoutTitle' => $this->trans(
+                    'Editing page category %name%',
```
```suggestion
                    'Editing category %name%',
```

I don't know. I like the "Editing CMS category", and "Editing page category" is similar to "Edit category for this particular page". Maybe "pages category"...?

Julievrz replied:
Daniel's suggestion is clear and easy to scan but it needs a more specific translation domain.

Hlavtox replied:
Decided to use page category

kpodemski commented on src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php:
```diff
@@ -321,6 +332,13 @@ public function viewAction(
             'transferGuestAccountForm' => $transferGuestAccountForm,
             'privateNoteForm' => $privateNoteForm->createView(),
+            'layoutTitle' => $this->trans(
+                'Customer %name%',
```
```suggestion
                'Customer: %name%',
```

I really could use a separator there to make it more readable. It's a global suggestion. Wdyt?

Julievrz replied:
It's a good idea but if we do it here, then we have to do it everywhere.

l-delin commented:
Hello @Hlavtox! 
First of all, thank you for your contributions. I see you proposed a lot of changes.
While we all know there’s a great need for unification in PrestaShop, there are some points I’m going to need you to clarify:

- There is a lot of edited/new wording here, but there isn’t enough information and context for me to understand a big part of it. I know translation domains are here for context, but still, it sometimes isn’t enough.
- These new wordings imply a lot of changes, especially to user doc, dev docs, and especially for translation and translators. While we try to keep docs up to date with new wordings, your changes will need us to update a big part of user resources. This is going to require a list of everything we need to change to try and help users as best we can.
- Therefore I ask: could you open a new issue with your new wording so that the content team can have a look at it? We need to lay down a blueprint of everything we’ll have to change and what it implies in order to edit your wording.
- We're gonna take a deep look at your propositions with all concerned stakeholders to identify what we can change or not according to the project's content team. 

Thank you! :-)

Hlavtox commented:
Hi @l-delin,

> There is a lot of edited/new wording here, but there isn’t enough information and context for me to understand a big part of it. I know translation domains are here for context, but still, it sometimes isn’t enough.

I added a list of all wordings to the PR description, these are all the pages that should be checked. I did not change all of the page titles, but just so we are unified.

I also added a list of the changed grid and form namings - they are all unified to singular or plural form.

> These new wordings imply a lot of changes, especially to user doc, dev docs, and especially for translation and translators. While we try to keep docs up to date with new wordings, your changes will need us to update a big part of user resources. This is going to require a list of everything we need to change to try and help users as best we can.

It's only a few words, don't worry. It has to be done at some point, when it was freestyle up until now.

> Therefore I ask: could you open a new issue with your new wording so that the content team can have a look at it? We need to lay down a blueprint of everything we’ll have to change and what it implies in order to edit your wording.

You have a list on the top, I won't open an issue for every wording.

> We're gonna take a deep look at your propositions with all concerned stakeholders to identify what we can change or not according to the project's content team.

Not sure I care about stakeholders, they are not paying for my work. They did not instruct any of their employees to do it up until now, so what is their position to tell me now what I should do when I spent 2 days unifying this? I did my best on this PR and I am sure that all users of this OPENSOURCE project will appreciate it. I graduated from translation and interpretation, so I think I did not do a bad job. ✌️ 

Regards

Julievrz commented on src/Core/Grid/Definition/Factory/BackupDefinitionFactory.php:
```diff
@@ -55,7 +55,7 @@ protected function getId()
     protected function getName()
     {
-        return $this->trans('DB Backup', [], 'Admin.Navigation.Menu');
+        return $this->trans('DB backups', [], 'Admin.Navigation.Menu');
```
You're right, there are several backups, so it should be plural. ✅

Julievrz commented on src/Core/Grid/Definition/Factory/CountryGridDefinitionFactory.php:
```diff
@@ -67,7 +67,7 @@ protected function getId(): string
     protected function getName(): string
     {
-        return $this->trans('Country', [], 'Admin.Global');
+        return $this->trans('Countries', [], 'Admin.Global');
```
Are you talking about the "Countries" tab in International > Locations?

<img width="1224" alt="Capture d’écran 2023-02-17 à 10 49 43" src="https://user-images.githubusercontent.com/70583503/219611062-30867f29-1508-414f-a08a-e8e44ab3bbd5.png">

It's already plural. Can you give more context, please?

Hlavtox replied:
We need to fix teh couintry number in brackets

EDIT: let somebody else fix it :D

Julievrz commented on src/Core/Grid/Definition/Factory/CustomerGridDefinitionFactory.php:
```diff
@@ -129,7 +129,7 @@ protected function getId()
     protected function getName()
     {
-        return $this->trans('Manage your Customers', [], 'Admin.Orderscustomers.Feature');
+        return $this->trans('Customers', [], 'Admin.Orderscustomers.Feature');
```
You're right, let's keep the same wording as in the menu for consistency. ✅

Julievrz commented on src/Core/Grid/Definition/Factory/CustomerThreadGridDefinitionFactory.php:
```diff
@@ -101,7 +101,7 @@ protected function getId(): string
     protected function getName(): string
     {
-        return $this->trans('Customer thread', [], 'Admin.Global');
+        return $this->trans('Customer threads', [], 'Admin.Global');
```
Where does this string appear?
We couldn't find it in the software or in Crowdin. I believe it's in the Customer Service section but I couldn't find it. If it's on a specific page such as the Customer Service one, then it should be in a more specific domain (probably Admin.Orderscustomers?)

Julievrz commented on src/Core/Grid/Definition/Factory/WebserviceKeyDefinitionFactory.php:
```diff
@@ -66,7 +66,7 @@ protected function getId()
     protected function getName()
     {
-        return $this->trans('Webservice', [], 'Admin.Navigation.Menu');
+        return $this->trans('Webservice keys', [], 'Admin.Navigation.Menu');
```
The first section of the page is about webservice keys. However, the second one deals with webservice in general. We should keep it general.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/AuthorizationServer/ApplicationController.php:
```diff
@@ -99,6 +99,7 @@ public function createAction(Request $request): Response
             'help_link' => $this->generateSidebarLink('AdminAuthorizationServer'),
             'enableSidebar' => true,
             'applicationForm' => $authorizedApplicationForm->createView(),
+            'layoutTitle' => $this->trans('New authorized application', 'Admin.Actions'),
```
We couldn't find the string, could you give us a bit more context please? 😌 It seems that it's a specific string on a specific page, so maybe it should be in a more specific domain.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/AuthorizationServer/ApplicationController.php:
```diff
@@ -137,6 +138,13 @@ public function editAction(Request $request, int $applicationId): Response
             'applicationForm' => $authorizedApplicationForm->createView(),
             'editableApplication' => $editableApplication,
+            'layoutTitle' => $this->trans(
+                'Editing application %application_name%',
```
Same here, we couldn't find the string, can you help please? ☺️

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/BackupController.php:
```diff
@@ -115,7 +115,7 @@ public function downloadViewAction(Request $request, $downloadFileName)
                 'size' => $backup->getSize(),
             ],
-            'layoutTitle' => $this->trans('View', 'Admin.Actions'),
+            'layoutTitle' => $this->trans('Downloading backup %s', 'Admin.Actions', [$downloadFileName]),
```
If I understand well, you want to add a text when hovering over the download icon? 
![image](https://user-images.githubusercontent.com/70583503/219621206-d9463c6d-33de-4379-ab0b-96822a95c1ce.png)
Or are you talking about one of these strings?
![image](https://user-images.githubusercontent.com/70583503/219621364-fed614ad-7df1-47ca-93c7-9e7e9f4ebe36.png)

In any case, I don't see what that's got to do with the "View" string in Admin.Actions which is used in many different places. If we replace this string we could end up with inconsistent wordings elsewhere. here's a,n example on the Orders page, which also uses this Admin.Actions's string: ![image](https://user-images.githubusercontent.com/70583503/219621635-02bdaf52-b101-4e12-9fe9-4adb05b565e4.png)

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmployeeController.php:
```diff
@@ -303,6 +304,7 @@ public function createAction(Request $request)
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'employeeForm' => $employeeForm->createView(),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New employee', 'Admin.Actions'),
```
You're right. On the product page, we have "New product" with a plus icon. But on all the other pages (files, categories, attributes and features, etc.) we have "Add a new..." and a plus icon. 
We should have consistent CTAs on all pages. But I think it would be easier to modify the one on the product page instead.

So we would have "Add a new product", "Add a new employee", "Add a new category", etc.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/EmployeeController.php:
```diff
@@ -390,6 +392,15 @@ public function editAction($employeeId, Request $request)
             'editableEmployee' => $editableEmployee,
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing employee %lastname% %firstname%',
```
```suggestion
                'Editing %firstname% %lastname%\'s profile',
```

Hlavtox replied:
✅ Done

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php:
```diff
@@ -146,7 +146,7 @@ public function createAction(Request $request)
         return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/Profiles/create.html.twig', [
             'profileForm' => $form->createView(),
-            'layoutTitle' => $this->trans('Add new profile', 'Admin.Advparameters.Feature'),
+            'layoutTitle' => $this->trans('New profile', 'Admin.Advparameters.Feature'),
```
Let's just modify the string on the product page so that we have "Add a new X" on all pages. ☺️

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php:
```diff
@@ -210,7 +210,7 @@ public function editAction($profileId, Request $request)
             'profileForm' => $form->createView(),
             'layoutTitle' => $this->trans(
-                'Edit: %value%',
+                'Editing profile %value%',
```
```suggestion
                'Editing %value%\'s profile',
```

Hlavtox replied:
Let's do Editing Translator profile

Hlavtox replied:
✅ Done

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php:
```diff
@@ -210,7 +210,7 @@ public function editAction($profileId, Request $request)
             'layoutTitle' => $this->trans(
-                'Edit: %value%',
+                'Editing profile %value%',
                 'Admin.Catalog.Feature',
```
```suggestion
                'Admin.Advparameters.Feature',
```
That's not your fault, the error was already there. But let's take the opportunity to correct it!

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php:
```diff
@@ -191,7 +191,7 @@ public function createAction(Request $request)
 
         return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/RequestSql/create.html.twig', [
-            'layoutTitle' => $this->trans('SQL Manager', 'Admin.Navigation.Menu'),
+            'layoutTitle' => $this->trans('New SQL request', 'Admin.Navigation.Menu'),
```
<img width="1226" alt="Capture d’écran 2023-02-17 à 12 02 15" src="https://user-images.githubusercontent.com/70583503/219630881-69ef727d-0bca-4996-9a10-2d2ea32b4aa9.png">

What string would you like to modify? 

The one in the tab, or the one in the CTA? 

Also, "SQL query" seems to be more common and is already used everywhere else on the page. 

What do you think?

Hlavtox replied:
✅ Changed to query

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php:
```diff
@@ -244,7 +244,7 @@ public function editAction(int $sqlRequestId, Request $request)
 
         return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/RequestSql/edit.html.twig', [
-            'layoutTitle' => $this->trans('SQL Manager', 'Admin.Navigation.Menu'),
+            'layoutTitle' => $this->trans('Editing SQL request %request%', 'Admin.Actions', ['%request%' => $sqlRequestForm->getData()['name']]),
```
Same question here, why not use "query"? Maybe you have a good reason, let me know! ☺️ 
Also, which string would be modified here exactly? Can you provide a screenshot, please? Just to be sure that the right translation domain is used, and that this won't have impacts elsewhere.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php:
```diff
@@ -344,7 +344,7 @@ public function viewAction(Request $request, int $sqlRequestId)
         return $this->render('@PrestaShop/Admin/Configure/AdvancedParameters/RequestSql/view.html.twig', [
             'layoutHeaderToolbarBtn' => [],
-            'layoutTitle' => $this->trans('SQL Manager', 'Admin.Navigation.Menu'),
+            'layoutTitle' => $this->trans('Result of SQL request', 'Admin.Navigation.Menu'),
```
Not sure I understand your suggestion. Can you elaborate?

Hlavtox replied:
✅ Changed to query

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php:
```diff
@@ -97,6 +97,7 @@ public function createAction(Request $request)
             '@PrestaShop/Admin/Configure/AdvancedParameters/Webservice/create.html.twig',
             [
                 'webserviceKeyForm' => $form->createView(),
+                'layoutTitle' => $this->trans('New webservice key', 'Admin.Actions'),
```
If we're talking about the CTA, let's keep "Add new webservice key" for consistency.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php:
```diff
@@ -135,6 +136,13 @@ public function editAction($webserviceKeyId, Request $request)
                 'webserviceKeyForm' => $form->createView(),
+                'layoutTitle' => $this->trans(
+                    'Editing webservice key %key%',
+                    'Admin.Actions',
```
```suggestion
                    'Admin.Advparameters.Feature',
```

Julievrz replied:
I think it's the wrong domain that is used as the string is specific to the Webservice page.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php:
```diff
@@ -148,6 +148,7 @@ public function createAction(Request $request)
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'contactForm' => $contactForm->createView(),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New contact', 'Admin.Actions'),
```
Same, I suggest keeping "Add new contact" for consistency.
<img width="1219" alt="Capture d’écran 2023-02-17 à 12 12 02" src="https://user-images.githubusercontent.com/70583503/219634434-3c3b39a7-a170-4931-a3e7-187f0597343b.png">

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php:
```diff
@@ -192,6 +193,13 @@ public function editAction($contactId, Request $request)
             'contactForm' => $contactForm->createView(),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing contact %name%',
```
```suggestion
                'Editing %name%',
```

Hlavtox replied:
✅

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php:
```diff
@@ -192,6 +193,13 @@ public function editAction($contactId, Request $request)
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing contact %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.Shopparameters.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -140,6 +140,7 @@ public function createAction(Request $request)
                 'Admin.Notifications.Info'
             ),
             'multistoreIsUsed' => $this->get('prestashop.adapter.multistore_feature')->isUsed(),
+            'layoutTitle' => $this->trans('New page configuration', 'Admin.Actions'),
```
If you're talking about the CTA on the SEO & URLs page, we should use a more specific domain such as Admin.Shopparameters.Feature 
See https://devdocs.prestashop-project.org/8/development/internationalization/translation/translation-domains/
Also, the CTA needs to match the wording we'll choose for the page's name (SEO & URLs)

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php:
```diff
@@ -175,6 +176,13 @@ public function editAction($metaId, Request $request)
         return $this->render('@PrestaShop/Admin/Configure/ShopParameters/TrafficSeo/Meta/edit.html.twig', [
             'meta_form' => $metaForm->createView(),
+            'layoutTitle' => $this->trans(
+                'Editing configuration for %name%',
```
Same, we need to use a more specific domain.

Hlavtox replied:
TODO wording

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php:
```diff
@@ -186,18 +189,28 @@ public function editAction(int $orderStateId, Request $request)
                 }, $this->get('prestashop.adapter.legacy.context')->getLanguages()),
+            'layoutTitle' => $this->trans(
+                'Editing status %name%',
+                'Admin.Actions',
```
But again, we should use a more specific domain. https://devdocs.prestashop-project.org/8/development/internationalization/translation/translation-domains/

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php:
```diff
@@ -235,6 +248,8 @@ public function createOrderReturnStateAction(Request $request)
             ),
             'multistoreIsUsed' => $this->get('prestashop.adapter.multistore_feature')->isUsed(),
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New return status', 'Admin.Actions'),
```
If it's a CTA, let's keep the "Add new X" structure for consistency + we should use a more specific domain than Admin.Actions

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php:
```diff
@@ -268,11 +283,21 @@ public function editOrderReturnStateAction(int $orderReturnStateId, Request $req
             'contextLangId' => $this->getContextLangId(),
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing return status %name%',
```
Needs a more specific domain

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/SearchEnginesController.php:
```diff
@@ -159,6 +159,13 @@ public function editAction(int $searchEngineId, Request $request): Response
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing search engine %name%',
```
Needs a more specific domain

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php:
```diff
@@ -232,6 +232,7 @@ public function createAction(Request $request)
                 'cmsCategoryParentId' => $categoryParentId,
                 'enableSidebar' => true,
                 'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+                'layoutTitle' => $this->trans('New page', 'Admin.Actions'),
```
Needs a more specific domain + keep the "Add new X" structure for consistency.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php:
```diff
@@ -309,6 +310,13 @@ public function editAction(Request $request, $cmsPageId)
                 'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
                 'previewUrl' => $previewUrl,
+                'layoutTitle' => $this->trans(
+                    'Editing page %name%',
```
Needs a more specific domain

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeController.php:
```diff
@@ -236,6 +236,7 @@ public function importAction(Request $request)
 
         return $this->render('@PrestaShop/Admin/Improve/Design/Theme/import.html.twig', [
             'importThemeForm' => $importThemeForm->createView(),
+            'layoutTitle' => $this->trans('Theme import', 'Admin.Actions'),
```
Can you provide a screenshot? If it's a CTA, it should be "Import theme". Also, it needs a more specific translation domain.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeController.php:
```diff
@@ -423,6 +424,7 @@ public function customizeLayoutsAction(Request $request)
         return $this->render('@PrestaShop/Admin/Improve/Design/Theme/customize_page_layouts.html.twig', [
             'pageLayoutCustomizationForm' => $pageLayoutCustomizationForm->createView(),
             'pages' => $pages,
+            'layoutTitle' => $this->trans('Choose layouts', 'Admin.Actions'),
```
Same, the domain can't be Admin.Actions if the CTA appears on a specific page.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php:
```diff
@@ -155,6 +155,7 @@ public function createAction(Request $request)
         return $this->render('@PrestaShop/Admin/Improve/International/Currency/create.html.twig', [
             'isShopFeatureEnabled' => $multiStoreFeature->isUsed(),
             'currencyForm' => $currencyForm->createView(),
+            'layoutTitle' => $this->trans('New currency', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new currency', 'Admin.International.Feature),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php:
```diff
@@ -101,6 +101,7 @@ public function createAction(Request $request): Response
             'countryForm' => $countryForm->createView(),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New country', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new country', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php:
```diff
@@ -194,6 +195,13 @@ public function editAction($currencyId, Request $request)
             'isShopFeatureEnabled' => $multiStoreFeature->isUsed(),
             'currencyForm' => $currencyForm->createView(),
+            'layoutTitle' => $this->trans(
+                'Editing currency %name%',
```
This string should be in Admin.International.Feature as it's specific to the International section.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php:
```diff
@@ -141,6 +142,7 @@ public function createAction(Request $request)
             'languageForm' => $languageForm->createView(),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New language', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new language', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/LanguageController.php:
```diff
@@ -200,6 +202,13 @@ public function editAction($languageId, Request $request)
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing language %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.International.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -179,7 +179,7 @@ public function editAction(int $stateId, Request $request): Response
         return $this->render('@PrestaShop/Admin/Improve/International/Locations/State/edit.html.twig', [
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Edit: %value%', 'Admin.Actions', ['%value%' => $editableState->getName()]),
+            'layoutTitle' => $this->trans('Editing state %value%', 'Admin.Actions', ['%value%' => $editableState->getName()]),
```
```suggestion
            'layoutTitle' => $this->trans('Editing state %value%', 'Admin.International.Feature', ['%value%' => $editableState->getName()]),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php:
```diff
@@ -223,6 +223,7 @@ public function createAction(Request $request): Response
                 'Admin.Notifications.Info'
             ),
             'multistoreIsUsed' => $this->get('prestashop.adapter.multistore_feature')->isUsed(),
+            'layoutTitle' => $this->trans('New state', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new state', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php:
```diff
@@ -190,6 +190,7 @@ public function createAction(Request $request)
                 'Admin.Notifications.Info'
             ),
             'multistoreIsUsed' => $this->get('prestashop.adapter.multistore_feature')->isUsed(),
+            'layoutTitle' => $this->trans('New tax', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new tax', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php:
```diff
@@ -247,6 +248,13 @@ public function editAction(Request $request, $taxId)
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing tax %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.International.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/TaxRulesGroupController.php:
```diff
@@ -109,6 +109,7 @@ public function createAction(Request $request): Response
             'enableSidebar' => true,
             'taxRulesGroupForm' => $taxRulesGroupForm->createView(),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'layoutTitle' => $this->trans('New tax rule', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new tax rule', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/TaxRulesGroupController.php:
```diff
@@ -150,7 +151,7 @@ public function editAction(Request $request, int $taxRulesGroupId): Response
         return $this->render('@PrestaShop/Admin/Improve/International/TaxRulesGroup/edit.html.twig', [
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Edit: %value%', 'Admin.Actions', ['%value%' => $taxRulesGroupForm->getData()['name']]),
+            'layoutTitle' => $this->trans('Editing tax rule %value%', 'Admin.Actions', ['%value%' => $taxRulesGroupForm->getData()['name']]),
```
```suggestion
            'layoutTitle' => $this->trans('Editing tax rule %value%', 'Admin.International.Feature', ['%value%' => $taxRulesGroupForm->getData()['name']]),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -136,6 +136,7 @@ public function createAction(Request $request): Response
             'zoneForm' => $zoneForm->createView(),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New zone', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new zone', 'Admin.International.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -189,6 +190,13 @@ public function editAction(int $zoneId, Request $request): Response
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Editing zone %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.International.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:
```diff
@@ -311,6 +311,7 @@ public function createAction(Request $request): Response
             'customerId' => $customerId,
             'customerInformation' => $customerInfo,
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New address', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new address', 'Admin.Orderscustomers.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/AttachmentController.php:
```diff
@@ -124,7 +124,7 @@ public function createAction(Request $request)
             'lightDisplay' => $request->query->has('liteDisplaying'),
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Add new file', 'Admin.Catalog.Feature'),
+            'layoutTitle' => $this->trans('New file', 'Admin.Catalog.Feature'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new file', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CatalogPriceRuleController.php:
```diff
@@ -214,6 +214,7 @@ public function createAction(Request $request): Response
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
             'catalogPriceRuleForm' => $catalogPriceRuleForm->createView(),
+            'layoutTitle' => $this->trans('New catalog rule', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new price rule', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CatalogPriceRuleController.php:
```diff
@@ -256,6 +257,13 @@ public function editAction(Request $request, int $catalogPriceRuleId): Response
             'catalogPriceRuleForm' => $catalogPriceRuleForm->createView(),
             'catalogPriceRuleName' => $editableCatalogPriceRule->getName(),
+            'layoutTitle' => $this->trans(
+                'Editing catalog rule %name%',
```
```suggestion
                'Editing price rule %name%',
```

Hlavtox replied:
Done

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CatalogPriceRuleController.php:
```diff
@@ -256,6 +257,13 @@ public function editAction(Request $request, int $catalogPriceRuleId): Response
             'catalogPriceRuleName' => $editableCatalogPriceRule->getName(),
+            'layoutTitle' => $this->trans(
+                'Editing catalog rule %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:
```diff
@@ -110,6 +110,11 @@ public function indexAction(Request $request, CategoryFilters $filters)
 
+        $layoutTitle = $this->trans('Categories', 'Admin.Navigation.Menu');
+        if (!$categoryViewData['is_home_category']) {
+            $layoutTitle = $this->trans('Category %name%', 'Admin.Actions', ['%name%' => $categoryViewData['name']]);
```
```suggestion
            $layoutTitle = $this->trans('Category %name%', 'Admin.Catalog.Feature', ['%name%' => $categoryViewData['name']]);
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:
```diff
@@ -193,6 +199,7 @@ public function createAction(Request $request)
                 'allowMenuThumbnailsUpload' => true,
                 'categoryForm' => $categoryForm->createView(),
                 'defaultGroups' => $defaultGroups,
+                'layoutTitle' => $this->trans('New category', 'Admin.Actions'),
```
```suggestion
                'layoutTitle' => $this->trans('Add new category', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:
```diff
@@ -241,6 +248,7 @@ public function createRootAction(Request $request)
                 'allowMenuThumbnailsUpload' => true,
                 'rootCategoryForm' => $rootCategoryForm->createView(),
                 'defaultGroups' => $defaultGroups,
+                'layoutTitle' => $this->trans('New category', 'Admin.Actions'),
```
```suggestion
                'layoutTitle' => $this->trans('Add new category', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:
```diff
@@ -326,6 +334,13 @@ public function editAction($categoryId, Request $request)
                     ->getUrl($categoryId, '{friendly-url}'),
+                'layoutTitle' => $this->trans(
+                    'Editing category %name%',
+                    'Admin.Actions',
```
```suggestion
                    'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:
```diff
@@ -399,6 +414,13 @@ public function editRootAction($categoryId, Request $request)
                     ->getUrl($categoryId, '{friendly-url}'),
+                'layoutTitle' => $this->trans(
+                    'Editing category %name%',
+                    'Admin.Actions',
```
```suggestion
                    'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php:
```diff
@@ -91,6 +91,7 @@ public function createAction(Request $request): Response
         if (!$this->isFeatureEnabled()) {
             return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
                 'showDisabledFeatureWarning' => true,
+                'layoutTitle' => $this->trans('New feature', 'Admin.Actions'),
```
```suggestion
                'layoutTitle' => $this->trans('Add new feature', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php:
```diff
@@ -114,6 +115,7 @@ public function createAction(Request $request): Response
 
         return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
             'featureForm' => $featureForm->createView(),
+            'layoutTitle' => $this->trans('New feature', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new feature', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php:
```diff
@@ -280,6 +282,13 @@ private function renderEditForm(array $parameters = [])
             'contextLangId' => $this->getConfiguration()->get('PS_LANG_DEFAULT'),
+            'layoutTitle' => $this->trans(
+                'Editing feature %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php:
```diff
@@ -160,6 +160,7 @@ public function createAction(Request $request)
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
             'manufacturerForm' => $manufacturerForm->createView(),
+            'layoutTitle' => $this->trans('New brand', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new brand', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php:
```diff
@@ -187,12 +188,18 @@ public function viewAction(Request $request, $manufacturerId)
             'enableSidebar' => true,
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'layoutTitle' => $this->trans(
+                'Brand %name%',
```
Can you give us more context, please? The string doesn't seem to be an action. It's probably in Admin.Catalog.Feature

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php:
```diff
@@ -241,6 +248,13 @@ public function editAction(Request $request, $manufacturerId)
             'logoImage' => $editableManufacturer->getLogoImage(),
+            'layoutTitle' => $this->trans(
+                'Editing brand %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php:
```diff
@@ -597,7 +611,7 @@ public function createAddressAction(Request $request)
         return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/Address/create.html.twig', [
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Add new address', 'Admin.Orderscustomers.Feature'),
+            'layoutTitle' => $this->trans('New brand address', 'Admin.Orderscustomers.Feature'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new brand address', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php:
```diff
@@ -651,7 +665,7 @@ public function editAddressAction(Request $request, $addressId)
         return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/Address/edit.html.twig', [
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Brands', 'Admin.Catalog.Feature'),
+            'layoutTitle' => $this->trans('Editing brand address #%id%', 'Admin.Orderscustomers.Feature', ['%id%' => $addressId]),
```
```suggestion
            'layoutTitle' => $this->trans('Editing brand address #%id%', 'Admin.Catalog.Feature', ['%id%' => $addressId]),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php:
```diff
@@ -129,6 +129,8 @@ public function createAction(Request $request)
         return $this->render('@PrestaShop/Admin/Sell/Catalog/Suppliers/add.html.twig', [
             'supplierForm' => $supplierForm->createView(),
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New supplier', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new supplier', 'Admin.Catalog.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php:
```diff
@@ -332,6 +334,13 @@ public function editAction(Request $request, $supplierId)
             'logoImage' => $editableSupplier->getLogoImage(),
+            'layoutTitle' => $this->trans(
+                'Editing supplier %name%',
+                'Admin.Actions',
```
```suggestion
                'Admin.Catalog.Feature',
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Catalog/SupplierController.php:
```diff
@@ -392,12 +401,18 @@ public function viewAction(Request $request, $supplierId)
             'enableSidebar' => true,
+            'layoutTitle' => $this->trans(
+                'Supplier %name%',
+                'Admin.Actions',
```
Can we have more context? This string doesn't seem to be an action. It should probably be in Admin.Catalog.Feature.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Customer/CustomerController.php:
```diff
@@ -192,6 +193,8 @@ public function createAction(Request $request)
             'displayInIframe' => $request->query->has('submitFormAjax'),
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('New customer', 'Admin.Actions'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new customer', 'Admin.Orderscustomers.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/CustomerThreadController.php:
```diff
@@ -116,7 +116,7 @@ public function viewAction(Request $request, int $customerThreadId)
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('View', 'Admin.Actions'),
+            'layoutTitle' => $this->trans('Customer thread #%s', 'Admin.Actions', [$customerThreadId]),
```
I added a comment above about the same issue

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php:
```diff
@@ -112,7 +112,7 @@ public function createAction(Request $request): Response
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
-            'layoutTitle' => $this->trans('Add new', 'Admin.Actions'),
+            'layoutTitle' => $this->trans('New message', 'Admin.Actions'),
```
Can you give more context, please? Thank you!

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/OrderMessageController.php:
```diff
@@ -167,7 +167,7 @@ public function editAction(int $orderMessageId, Request $request): Response
             'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
             'enableSidebar' => true,
-            'layoutTitle' => sprintf($this->trans('Edit: %s', 'Admin.Actions'), $orderMessageName),
+            'layoutTitle' => $this->trans('Editing message %s', 'Admin.Actions', [$orderMessageName]),
```
Same here, it will us find the right translation domain. Thanks!

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php:
```diff
@@ -90,7 +90,7 @@ public function viewAction(Request $request, $cartId)
         return $this->render('@PrestaShop/Admin/Sell/Order/Cart/view.html.twig', [
             'cartView' => $cartView,
-            'layoutTitle' => $this->trans('View', 'Admin.Actions'),
+            'layoutTitle' => $this->trans('Cart #%s', 'Admin.Actions', [$cartId]),
```
We need more context. The "View" string (Admin.Actions) is used in other places. It shouldn't be modified.

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:
```diff
@@ -263,6 +263,7 @@ public function createAction(Request $request)
             'giftSettingsEnabled' => (bool) $configuration->get('PS_GIFT_WRAPPING'),
             'stockManagementEnabled' => (bool) $configuration->get('PS_STOCK_MANAGEMENT'),
             'isB2BEnabled' => (bool) $configuration->get('PS_B2B_ENABLE'),
+            'layoutTitle' => $this->trans('New manual order', 'Admin.Orderscustomers.Feature'),
```
```suggestion
            'layoutTitle' => $this->trans('Add new order', 'Admin.Orderscustomers.Feature'),
```

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Employee/Blocks/form.html.twig:
```diff
@@ -31,7 +31,7 @@
   <div class="card">
     <h3 class="card-header">
-      {{ 'Employees'|trans }}
+      {{ 'Employee'|trans }}
```
The string is already plural, or you're talking about something else?
<img width="1224" alt="Capture d’écran 2023-02-17 à 14 08 50" src="https://user-images.githubusercontent.com/70583503/219661442-f8f577ce-3f00-4e75-9f34-e9c51726b5ee.png">

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/Webservice/Blocks/form.html.twig:
```diff
@@ -29,7 +29,7 @@
   <div class="card">
     <h3 class="card-header">
-      {{ 'Webservice Accounts'|trans({}, 'Admin.Advparameters.Feature') }}
+      {{ 'Webservice key'|trans({}, 'Admin.Advparameters.Feature') }}
```
Yep, agreed! ✅

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/Contact/Contacts/Blocks/form.html.twig:
```diff
@@ -30,7 +30,7 @@
       <h3 class="card-header">
         <i class="material-icons">mail_outline</i>
-        {{ 'Contacts'|trans({}, 'Admin.Shopparameters.Feature') }}
+        {{ 'Contact'|trans({}, 'Admin.Shopparameters.Feature') }}
```
Good catch! ☺️

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/OrderReturnStates/Blocks/form.html.twig:
```diff
@@ -30,7 +30,7 @@
       <h3 class="card-header">
         <i class="material-icons">schedule</i>
-        {{ 'Return statuses'|trans({}, 'Admin.Shopparameters.Feature') }}
+        {{ 'Return status'|trans({}, 'Admin.Shopparameters.Feature') }}
```
There could have more than 1 status, don't you think? 

Or am I missing something?

<img width="1214" alt="Capture d’écran 2023-02-17 à 14 14 32" src="https://user-images.githubusercontent.com/70583503/219662352-c39dfebe-0fc7-4a76-9e1e-39a212036e61.png">

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/TrafficSeo/Meta/Blocks/form.html.twig:
```diff
@@ -27,9 +27,9 @@
-      {{ 'Meta tags'|trans({}, 'Admin.Shopparameters.Feature') }}
-    </div>
+    <h3 class="card-header">
+      {{ 'Page configuration'|trans({}, 'Admin.Shopparameters.Feature') }}
```
Can you give more context, please?

Hlavtox replied:
TODO wording

Julievrz commented on src/PrestaShopBundle/Resources/views/Admin/Configure/ShopParameters/maintenance.html.twig:
```diff
@@ -33,7 +33,7 @@
       <h3 class="card-header">
         <i class="material-icons">business_center</i>
-        {{ 'General'|trans({}, 'Admin.Global') }}
+        {{ 'Maintenance'|trans({}, 'Admin.Global') }}
```
Yep, you're right
<img width="1218" alt="Capture d’écran 2023-02-17 à 14 17 49" src="https://user-images.githubusercontent.com/70583503/219663017-301899c4-6a2e-422d-9b3f-ae802a1b5725.png">

Julievrz reviewed (commented):
Hello Daniel, 
Thank you for your contribution. I agree with you that there are some wording inconsistencies. 
Léa and I had a closer look at it. Your PR will help bring more consistency to the project. However, we have to be careful with the [translation domains](https://devdocs.prestashop-project.org/8/development/internationalization/translation/translation-domains/) used. We added comments when necessary. 
We spent time checking each wording in PrestaShop but some of them lacked context. Your help would be appreciated, thank you.

Hlavtox committed:
Unify domains of page titles

Hlavtox committed:
Batch of corrections

Hlavtox committed:
Work batch

Hlavtox commented:
Checklist after a call with @Julievrz

### Finished
- [ ] Unify all page titles into single domain `Admin.Navigation.Menu`.
- [ ] Check domains of forms - they should be in a domain depending on a topic.
- [ ] Unify wording of order status to STATUS and STATUSES.
- [ ] Unify wording of CMS stuff to just Page and Category.
- [ ] SQL request => SQL query.
- [ ] Minor tweaks like `SQL query results,` `Editing Webmaster contact`, `Editing brand address`, `New order message`, `New order`
- [ ] Customer thread #ID - moved id into placeholder for better control and more context
- [ ] Check all CTAs in the header.
- [ ] `Add new theme` or `Import theme` CTA?
- [ ] Check wordings of form headers.

### To think about and apply
- Wording for a page configuration section in SEO & URLS (meta controller). Do we have better wording than `Page configuration`? It could be called `Meta configuration,` but it's not just about meta, it's also page title and it's URL.
- Think about the 2nd+ capital letters in names. `Tax rules` or `Tax Rules`?

Hlavtox committed:
Work batch

Julievrz commented:
> 
> ### What I found more that needs discussion
> * **Positions page**
>   
>   * Page title - I would go for `Module positions`.

**It makes more sense indeed.**

>   * CTA `Transplant a module` - It has a weird name and no icon. What about `Hook a module` and good old anchor icon? You transplant a kidney not a module. 😄

**I agree with the wording. I'll let @ElodieOS answer about the icon.**

> * **SEO & URLS page**
>   
>   * CTA `Add a new page` is weird, but I have no other idea. Maybe `Add new page configuration`?
>   * Wording for a meta list. Do we have better wording than `Pages configuration`? It could be called `Meta configuration,` but it's not just about meta, it's also page title and it's URL. We could go for `Page meta & URL configurations` or something like this?

>   * Wording for editing a page meta - Went for `Editing configuration for %name%`.

**"Page configuration" is already used in the Design section of the back office. What about "SEO settings"?**

**About the CTA "Add new page": based on my previous comment, what about "Set up new page"? This would echo the CTA of the "Improve your SEO" card. (We would need to replace "Configure" with "Set up", but that may be out of scope).** 

> * **Product page**
>   
>   * Should `New product` CTA be `Add new product`?

**Yes, for consistency, thank you.** 

> * **Themes page**
>   
>   * `Add new theme` or `Import theme` CTA? The page is about importing a theme.

**You're right, "Import theme" is more transparent.**

> * Think about the 2nd+ capital letters in names. `Tax rules` or `Tax Rules`?

**There's a rule in English saying that we should capitalize (almost) every word in a title. But it's more about articles, books, and songs. In PrestaShop, I suggest capitalizing every word constituting the navigation menu labels in the sidebar (for consistency) but favor sentence case inside pages. Sentence case will make the pages easier to "scan".** 

> ![positions](https://user-images.githubusercontent.com/6097524/219702126-66cb3405-4f55-46e4-a1df-554c64f1ea27.png)

**PS: Please note @l-delin is the wording manager for the project now and she should have the last word. I will let her make a final review when she comes back next week.** ☺️

Hlavtox committed:
Minor improvements

Hlavtox committed:
Implement feedback for meta page

Hlavtox commented:
All feedback resolved and updated previous comment. It will need some test adjustments, for example because of changing `New & Experimental Features` to `New & experimental features`.

Hlavtox committed:
Address feedback and unify grid names

Hlavtox committed:
Add titles to all missing controllers

Hlavtox committed:
Fix tests

Hlavtox commented:
@l-delin @Julievrz I have addressed your feedbacks from here and Slack, putting the Wording ✅ label.

l-delin commented:
Hello, I'm removing the label so I can go over the PR one more time and make sure everything is ok.
I will add it back then. Please refrain from doing this next time. Thank you

Hlavtox commented:
@l-delin OK, please do it in a reasonable timeframe and put it back so we can go on. :-)

l-delin commented:
Wording is all good to go. Thanks. ✅

0x346e3730 reviewed (approved):
lgtm

jolelievre commented on controllers/admin/AdminModulesPositionsController.php:
```diff
@@ -254,12 +254,6 @@ public function initContent()
 
     public function initPageHeaderToolbar()
     {
-        $this->page_header_toolbar_btn['save'] = [
```
Why did you remove this button? It's not a wording modification here it's a functional one

Hlavtox replied:
That button should not be there, it only refreshes the page.

jolelievre commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php:
```diff
@@ -356,6 +364,7 @@ protected function renderPage(Request $request, WebserviceKeyFilters $filters, F
                 'grid' => $this->presentGrid($grid),
                 'configurationWarnings' => $this->lookForWarnings(),
                 'webserviceStatus' => $this->getWebServiceStatus($request),
+                'enableSidebar' => true,
```
This is also a functional change

Hlavtox replied:
I only moved it from template to controller, so it's united.

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Improve/Design/Cms/Blocks/category_form.html.twig:
```diff
@@ -29,9 +29,9 @@
 
   <div class="card">
-    <div class="card-header">
-      {{ 'CMS Category'|trans({}, 'Admin.Design.Feature') }}
```
Why not keep `CMS category` here? It allows telling the difference with the other Category page (used for Product category)

Hlavtox replied:
Already validated with the product team.

jolelievre commented on src/PrestaShopBundle/Resources/views/Admin/Improve/Design/Theme/import.html.twig:
```diff
@@ -22,24 +22,9 @@
 {% form_theme importThemeForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}
 
-{% set layoutTitle = 'Theme & Logo'|trans({}, 'Admin.Navigation.Menu') ~ ' > ' ~ 'Theme'|trans({}, 'Admin.Design.Feature') %}
-{% set layoutHeaderToolbarBtn = {
-  'add': {
-    'href': path('admin_themes_import'),
-    'desc': 'Add new theme'|trans({}, 'Admin.Design.Feature'),
-    'icon': 'add'
-  },
-  'export': {
-    'href': path('admin_themes_export_current'),
-    'desc': 'Export current theme'|trans({}, 'Admin.Design.Feature'),
-    'icon': 'cloud_download'
-  }
-} %}
```
This changes the layout of the page sicne it removes two buttons no?

Hlavtox replied:
They are extra buttons on a subpage, import will get you to the same exact page.

jolelievre commented on src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/FeatureFlagController.php:
```diff
@@ -93,7 +93,7 @@ public function indexAction(Request $request): Response
             'enableSidebar' => true,
             'layoutHeaderToolbarBtn' => [],
-            'layoutTitle' => $this->trans('New & Experimental Features', 'Admin.Advparameters.Feature'),
+            'layoutTitle' => $this->trans('New & experimental features', 'Admin.Navigation.Menu'),
```
If you change layout title of pages accessible from the menu you should also change the default install data, or the many and the page content won't match

Hlavtox replied:
It changes the title only, not the menu. No install or upgrade is needed.

jolelievre reviewed (dismissed):
Thanks @Hlavtox

nesrineabdmouleh reviewed (commented):
UI tests => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/4352412719

florine2623 reviewed (commented):
Hello @Hlavtox ,

Thanks for the PR !

I have found Errors and Exceptions 😓 
I have enabled all Feature flags.

After creating a Cart Rule (and deleting it) I have an Exception in Catalog > Products > Edit an existing product :
<img width="1055" alt="Screenshot 2023-03-07 at 16 53 24" src="https://user-images.githubusercontent.com/16019289/223475576-f907f44d-39f9-4f5c-94c7-94be6ce65fb0.png">


In Catalog > Attributes & Features > Attributes, the "add new feature" button :
<img width="1230" alt="Screenshot 2023-03-07 at 16 42 43" src="https://user-images.githubusercontent.com/16019289/223472708-68e6f5d9-668d-4683-b866-51866c19e0da.png">

The Edit button doesn't work :
<img width="1220" alt="Screenshot 2023-03-07 at 16 45 27" src="https://user-images.githubusercontent.com/16019289/223473468-3f20c06b-73e7-495a-8f96-0fc54475f9a7.png">

In Catalog > Attributes & Features > Features, the "add new feature value" button :

<img width="1230" alt="Screenshot 2023-03-07 at 16 43 33" src="https://user-images.githubusercontent.com/16019289/223472925-8864ebe9-b6ab-4938-9c28-8bfcd60543fa.png">

The Edit button doesn't work :
<img width="1223" alt="Screenshot 2023-03-07 at 16 46 30" src="https://user-images.githubusercontent.com/16019289/223473723-eacb3e43-d2bf-4378-9472-12616921fc06.png">

In Catalog > Discount > Cart rule, the Add button is missing :
<img width="1232" alt="Screenshot 2023-03-07 at 16 51 15" src="https://user-images.githubusercontent.com/16019289/223474929-da2cb841-712e-481c-9d0d-cd20d71ee1a6.png">

In Customer Service, the page is empty :
<img width="1231" alt="Screenshot 2023-03-07 at 16 56 38" src="https://user-images.githubusercontent.com/16019289/223476570-b9598b07-a7fb-4bdb-866b-7c221c2432ef.png">

In Design Position, I just need to make sure this is the wanted behavior :
Should it be Module positions in both pages ?

https://user-images.githubusercontent.com/16019289/223478391-53c9b341-0407-40c5-983b-e58b69194099.mov

In International > Locations > Countries :
In the Call prefix, we are missing the + sign and the Enable coloumn is wrong.
<img width="1217" alt="Screenshot 2023-03-07 at 17 07 59" src="https://user-images.githubusercontent.com/16019289/223479892-15ce4345-c414-47f5-aa8e-7e1953a9ed57.png">

In International > Taxes, the Actions buttons are missing :
<img width="1219" alt="Screenshot 2023-03-07 at 17 11 29" src="https://user-images.githubusercontent.com/16019289/223480551-70e018c9-f7a8-4bc5-977c-eae6edc70184.png">

In General Parameters > Order Settings > Statuses, the columns ID and Name are unbalanced :
<img width="1224" alt="Screenshot 2023-03-07 at 17 13 25" src="https://user-images.githubusercontent.com/16019289/223481354-c0e1ad24-f60c-4b89-895b-da386fb245dd.png">

Add new order status page is broken :
<img width="1073" alt="Screenshot 2023-03-07 at 17 15 05" src="https://user-images.githubusercontent.com/16019289/223481753-0c90a90c-6a58-46b5-81a9-59c9d1a1f8e5.png">

Edition order status page is broken :
<img width="1054" alt="Screenshot 2023-03-07 at 17 15 43" src="https://user-images.githubusercontent.com/16019289/223481969-da6cd23d-8149-45a2-9dc5-4bef052dfcc5.png">

Edition of Return statuses color field is broken :
<img width="1216" alt="Screenshot 2023-03-07 at 17 16 23" src="https://user-images.githubusercontent.com/16019289/223482469-6e3771da-3e04-40bc-8e27-6a562f5fbe2d.png">

Shop Parameters > Customer Settings > Add new title page is broken :
<img width="660" alt="Screenshot 2023-03-07 at 17 18 16" src="https://user-images.githubusercontent.com/16019289/223483034-98298014-9aa7-40f3-8f92-8e451a0f7840.png">

Shop Parameters > Customer Settings > Edition of Title is broken :
<img width="1077" alt="Screenshot 2023-03-07 at 17 18 56" src="https://user-images.githubusercontent.com/16019289/223483246-47068df3-2fdc-4fe9-98a7-36970e52cc97.png">

In Advanced Parameters > Teams > Permissions Page :
<img width="1227" alt="Screenshot 2023-03-07 at 16 36 33" src="https://user-images.githubusercontent.com/16019289/223471222-6129e884-2131-4bf9-90e3-5ec6a69f012b.png">

Waiting for your feedback :)
Thanks a lot !!

kpodemski commented:
Hello @florine2623 

Almost all the errors you reported are reproducible using **the clean version of the develop branch** We shouldn't ask Daniel to fix those issues if they are already there and don't come from this PR.

I think you should have two installations while testing:
- one should be based on the PR you test
- the other one should be the clean version of the branch PR targets

If you have any errors with PR, you should immediately check if they are also there on the clean installation.  

I suggest retesting this PR and reporting things that are not related to this PR in separate issues, but if it's worth testing feature flags, that's the question to @Robin-Fischer-PS

Robin-Fischer-PS commented:
Hi there !

It's indeed worth testing feature flags. However, like you said @kpodemski , we usually test with 2 installs (one with PR/one without).

I change the label to Waiting for QA again, to check if the issues are indeed reproducible without PR.

Thanks all :)

florine2623 reviewed (dismissed):
Hello @kpodemski ,

I've checked everything, and indeed all the bugs reported exist on develop branch with all feature flags enabled. Thanks for letting me know.

I can validate this PR ^^ It is QA ✅ 
Good job @Hlavtox !

florine2623 commented:
**PLEASE DO NOT MERGE THIS RIGHT NOW ! @PrestaShop/prestashop-core-developers 

THE AUTO TESTS ARE BROKEN FOR THE MOMENT !**

@nesrineabdmouleh is working on it !

0x346e3730 reviewed (dismissed):
Trying to block the PR from being merged accidentally following florine's comment

nesrineabdmouleh committed:
Adapt BO-O2-O3-O4-05 tests to the new page names

nesrineabdmouleh committed:
Adapt BO-O8-11-13-14-15 tests to the new page names

nesrineabdmouleh committed:
Adapt FO and regression tests to the new page names

nesrineabdmouleh committed:
Update page names

nesrineabdmouleh commented:
UI tests => https://github.com/nesrineabdmouleh/testing_pr/actions/runs/4368532822

nesrineabdmouleh committed:
Fix lint errors

Hlavtox commented:
@kpodemski @0x346e3730 Approve and we can merge ❤️

Thank you @nesrineabdmouleh! 🚀

florine2623 commented:
@Hlavtox ,

It is not yet ready to merge. 
It's better to wait for @nesrineabdmouleh 's approval before merging  🙏 

Thanks all !

Hlavtox commented:
@florine2623 @nesrineabdmouleh Sorry, thought it's ready 🙃

nesrineabdmouleh committed:
Adapt BO>View carts test to the new page names

Hlavtox commented:
@nesrineabdmouleh It's just because it's not translated yet. It's not a regression.

![Snímek obrazovky 2023-03-09 114014](https://user-images.githubusercontent.com/6097524/223999785-313ef380-8ad9-497f-9475-bf59a0638ae2.jpg)

florine2623 reviewed (commented):
I have reproduced the same issue as Nesrine with the PR and only with the PR 😅 
<img width="1433" alt="Screenshot 2023-03-09 at 11 59 07" src="https://user-images.githubusercontent.com/16019289/224004437-ed188ffb-479e-43c9-8d93-99a535cfa424.png">

"Credit slips" should be in French "Avoirs".

Some others pages also have the issue like :
- Delivery Slips -> Bons de livraison
<img width="622" alt="Screenshot 2023-03-09 at 12 01 52" src="https://user-images.githubusercontent.com/16019289/224004712-6061f20b-fa55-4c63-a93f-bc374acefc9e.png">
- Order messages -> Messages prédéfinis
<img width="612" alt="Screenshot 2023-03-09 at 12 03 08" src="https://user-images.githubusercontent.com/16019289/224004974-1a62e261-7626-43b8-92f4-0af682911d41.png">
- Module Manager -> Gestionnaire de modules
<img width="622" alt="Screenshot 2023-03-09 at 12 04 28" src="https://user-images.githubusercontent.com/16019289/224005229-daea321e-03ba-468d-87d4-262dee7e5298.png">
- Email Theme -> Thème d'email
<img width="539" alt="Screenshot 2023-03-09 at 12 04 48" src="https://user-images.githubusercontent.com/16019289/224006009-c789bc6a-d25f-4e36-9f9b-54515f6fe391.png">
- Payment Methods -> Modes de paiement
<img width="615" alt="Screenshot 2023-03-09 at 12 05 24" src="https://user-images.githubusercontent.com/16019289/224006075-a4bb0c8e-6286-448c-adcb-683e1b80801f.png">
- Products Settings -> Produits
<img width="624" alt="Screenshot 2023-03-09 at 12 06 06" src="https://user-images.githubusercontent.com/16019289/224006334-085817f4-fcde-476c-af5e-d8407c5ba86e.png">
- Customer Settings -> Clients
<img width="532" alt="Screenshot 2023-03-09 at 12 06 13" src="https://user-images.githubusercontent.com/16019289/224006416-bc383ad1-3869-4f92-ae0e-044f3ea058db.png">
- SQL Manager and SQL queries -> Gestionnaire SQL
<img width="686" alt="Screenshot 2023-03-09 at 12 07 28" src="https://user-images.githubusercontent.com/16019289/224006689-c7369859-afd9-4806-9c8a-43a57251380b.png">

Could you check ?
Thanks 🤗

Hlavtox commented:
@florine2623 It seems it's the first PR that you are testing... When there is a new string, it's not translated until the version is getting released.

kpodemski commented:
I can confirm. @florine2623 @Robin-Fischer-PS, please make sure to document it somewhere. If the wording has changed, translations are not updated until before the release during catalog extraction.

boubkerbribri commented:
> I can confirm. @florine2623 @Robin-Fischer-PS, please make sure to document it somewhere. If the wording has changed, translations are not updated until before the release during catalog extraction.

The problem for us here, that Automatic tests will be useless here because the will display failed tests that are not really bugs (Analyzing each time will be hell). So id we merge like this, we will not use them to validate any PR before the release.
So we have 3 solutions, and you (maintainers + qa-func + qa-autom) tell me what the best solution : 
- Merge this PR and not use automatic tests for the rest of PRs (until the Release of 9.0.0).
- Do a catalog extraction on another PR right now (exception)
- Wait on this PR and merge it when we decide to release and do the catalog extraction.
WDYT ?

kpodemski commented:
@boubkerbribri 

>Merge this PR and not use automatic tests for the rest of PRs (until the Release of 9.0.0).

This would be hell. What about all the PRs that rely solely on the auto tests because there's no point in having QA testing them?

> Do a catalog extraction on another PR right now (exception)

I'll ask @jolelievre about his opinion there because I need to figure out how the process could look like.

> Wait on this PR and merge it when we decide to release and do the catalog extraction.
WDYT ?

This could be a massive problem in the end because rebasing it would be a nightmare after all changes in the meantime.

I don't see any reason to check the French version of wording in the automatic tests 🤷🏻 I understand matching original phrases, but translations...? Maybe tests for translations could be run on-demand?

From what I know, it's been considered a bad practice precisely for that reason, but aside from that, the problem here is that we check for translations.

Hlavtox commented:
I don't understand why suddenly after many years there is a problem in a PR that introduces new translations.

nesrineabdmouleh commented:
Hello @florine2623, @boubkerbribri, @Hlavtox 

I have a solution to solve the problem in UI tests, I can update the failed test => (check only with English language) ;)

What do you think ?

boubkerbribri commented:
> Hello @florine2623, @boubkerbribri, @Hlavtox, @kpodemski
> 
> I have a solution to solve the problem in UI tests, I can update the failed test => (check only with English language) ;)
> 
> What do you think ?

Works for me

Hlavtox commented:
@nesrineabdmouleh @boubkerbribri Another solution would be to insert the translation manually to `ps_translation` table when itializing the data for tests. Is it possible? This would allow you to check if translations are working, but you won't have to wait for translations to be in the XLF file.

nesrineabdmouleh committed:
Add missing types

nesrineabdmouleh committed:
Refacto BO>Header>My profile tests

kpodemski commented:
@Hlavtox I don't see how that would help, if there are new phrases we'd need to wait for translated version.

@boubkerbribri I don't see how exporting catalog would help there. It would require catalog sync + translations. It would slow down the process drastically. Changing wording shouldn't be blocked by the tests, or the fact that we need something to be translated to French. Still, I'd like to see @jolelievre opinion on this one.

Hlavtox commented:
@boubkerbribri I think something like this is planned - https://github.com/PrestaShop/PrestaShop/issues/31197

@eternoendless Boss, will it be automatic on like a nightly basis?

nesrineabdmouleh commented:
Finally UI tests :heavy_check_mark: 
The failures are known bugs 
https://github.com/nesrineabdmouleh/testing_pr/actions/runs/4368532822

Thank you all :)

kpodemski commented:
Thanks Daniel. Huge work. Thanks @nesrineabdmouleh @l-delin and @Julievrz for helping.

Hlavtox commented:
Yep, thanks everyone! @nesrineabdmouleh @boubkerbribri @l-delin @Julievrz & @florine2623 🎉

Now we need to be the protectors of wording when the rest is migrated ⚔️ :trollface:
